### PR TITLE
all: handle volt unit for electrification in data only

### PR DESCRIPTION
**Pull request by bougue-pe**
Front does not add 'V' as unit anymore (but some hard-coded front processing is not tackled and just migrated, see #6235 for a part of it).
The responsibility for unit is on the data side (rolling stock, catenaries, electrical profiles).

Hand-tested (cards, edit and curves on rolling-stock, legend and catenaries on map, traction on study's speed-space graph, simple stdcm use):
- OK to load a dump, migrate and use (prints 1500V and colors)
- OK to load a dump and use without migration, so with identical "wrong" values everywhere (front crashes on speed-space graph as expected for traction without 'V', otherwise prints 1500 black)

`osrd` part of https://github.com/osrd-project/osrd-dags/issues/135 (`osrd-dags` not migrated yet, both should be merged at the same time once done).

### CODE REVIEW

**Review by shenriotpro — APPROVED** [review]
OK for tests/

File: front/public/locales/fr/simulation.json [hunk 1]
```diff
@@ -31,10 +31,10 @@
     "incompatible": "Restriction de puissance incompatible"
   },
   "electricalProfiles": {
-    "25000": "25000V",
-    "1500": "1500V",
-    "15000": "15000V 16/2/3",
-    "3000": "3000V",
+    "25000V": "25000V",
+    "1500V": "1500V",
+    "15000V": "15000V 16/2/3",
```
Comment (Castavo) [inline front/public/locales/fr/simulation.json]: I feel like this hardcoding could be removed by also changing the import of "15000V" to "15000V 16/2/3"

File: front/public/locales/fr/translation.json [hunk 2]
```diff
@@ -254,7 +254,7 @@
         "catenaries": "Électrification",
         "catenary-default": "Électrification par défaut",
         "label": "Outil \"Électrification\"",
-        "voltage": "{{voltage}}V"
+        "voltage": "{{voltage}}"
```
Comment (Castavo) [inline front/public/locales/fr/translation.json]: Isn't this field useless now ?

**Review by Castavo — APPROVED** [review]
Works fine !

I ran the migration successfully, the run time with our current dump seems very acceptable.

I tested the cartography, RS editor and Electrification in the GEV, all seem to work fine after the migration.  
I did not check before migration.

File: front/public/locales/fr/translation.json (hunk 2 above)
Comment (clarani) [reply front/public/locales/fr/translation.json]: I agree, I think we can remove it

File: front/public/locales/en/simulation.json [hunk 3]
```diff
@@ -31,10 +31,10 @@
     "incompatible": "Incompatible power restriction "
   },
   "electricalProfiles": {
-    "25000": "25000V",
-    "1500": "1500V",
-    "15000": "15000V 16/2/3",
-    "3000": "3000V",
+    "25000V": "25000V",
+    "1500V": "1500V",
```
Comment (clarani) [inline front/public/locales/en/simulation.json]: We can remove those translations (and also 3000V)

File: front/public/locales/en/translation.json [hunk 4]
```diff
@@ -254,7 +254,7 @@
         "catenaries": "Electrification",
         "catenary-default": "Default catenary",
         "label": "Catenary tool",
-        "voltage": "{{voltage}}V"
+        "voltage": "{{voltage}}"
```
Comment (clarani) [inline front/public/locales/en/translation.json]: To remove here too

**Review by clarani — APPROVED** [review]
LGTM, tested with the migration :white_check_mark: 
Tested the rs editor, the infra editor, the reference map, the rs selector and the gev

File: front/public/locales/fr/simulation.json (hunk 1 above)
Comment (bougue-pe) [reply front/public/locales/fr/simulation.json]: I'll remove as much as possible for now, but the question is not decided and deferred to https://github.com/osrd-project/osrd-dags/issues/135#issuecomment-1876857154
So keeping the hard-coding for `15000V` -> `15000V 16/2/3`

File: front/public/locales/fr/translation.json (hunk 2 above)
Comment (bougue-pe) [reply front/public/locales/fr/translation.json]: Done in https://github.com/osrd-project/osrd/pull/6229/commits/8adfa05b0aac95d281834018901ec8e3f8b264e1

File: front/public/locales/fr/simulation.json (hunk 1 above)
Comment (bougue-pe) [reply front/public/locales/fr/simulation.json]: It looks like this was dead-code before anyway, so I removed and hand-tested study and stdcm in https://github.com/osrd-project/osrd/commit/8adfa05b0aac95d281834018901ec8e3f8b264e1
Maybe more tests would be nice (if you have an idea of its usage).

File: front/public/locales/en/translation.json (hunk 4 above)
Comment (bougue-pe) [reply front/public/locales/en/translation.json]: Done in https://github.com/osrd-project/osrd/commit/8adfa05b0aac95d281834018901ec8e3f8b264e1

File: front/public/locales/en/simulation.json (hunk 3 above)
Comment (bougue-pe) [reply front/public/locales/en/simulation.json]: Done, for details please check reply in https://github.com/osrd-project/osrd/pull/6229#discussion_r1442622039

File: editoast/migrations/2023-12-26-152659_add_electrification_unit/up.sql [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    SELECT jsonb_agg(
+        CASE WHEN elem->'electrificationUsage'->>'mode' ~ '^\d+$' AND elem->'electrificationUsage'->>'object_type' = 'Electrified'
+        THEN jsonb_set(elem, '{electrificationUsage, mode}', to_jsonb(elem->'electrificationUsage'->>'mode' || 'V'), FALSE)
+        ELSE elem END
+    )
+    FROM jsonb_array_elements(electrification_ranges) elem
+)
+WHERE electrification_ranges IS NOT NULL AND electrification_ranges != '[]'::jsonb;
+
+UPDATE infra SET "version" = (NULLIF("version", '0')::int + 1)::text;
```
Comment (Tguisnet) [inline editoast/migrations/2023-12-26-152659_add_electrification_unit/up.sql]: not useful

**Review by Tguisnet — CHANGES_REQUESTED** [review]

File: editoast/migrations/2023-12-26-152659_add_electrification_unit/up.sql (hunk 5 above)
Comment (bougue-pe) [reply editoast/migrations/2023-12-26-152659_add_electrification_unit/up.sql]: Removed: https://github.com/osrd-project/osrd/pull/6229/commits/86ec961e505544f08eceb15cd7421f38a0e417c7

**Review by Tguisnet — APPROVED** [review]
Lgtm

**Review by Erashin — APPROVED** [review]
Core part lgtm.

**Review by Castavo — APPROVED** [review]
I checked with the last data dump, works fine !

I tested map, RS editor and simulation with electrical profiles.

**Pull request merged**